libs/ui-components/src/utils/patch.ts

```diff
@@ -200,3 +206,54 @@ export const getApplicationPatches = (
 
   return patches;
 };
+
+export const getMicroShiftHookPatches = (
+  currentHooks: DeviceHooksSpec | undefined,
+  registerMicroShift: boolean,
+  basePath: string,
+) => {
+  const patches: PatchRequest = [];
+  if (registerMicroShift) {
+    if (!currentHooks) {
+      patches.push({
+        op: 'add',
+        path: `${basePath}/hooks/`,
+        value: {
+          afterUpdating: [ACMCrdHook, ACMImportHook],
+        },
+      });
+    } else if (!currentHooks.afterUpdating) {
+      patches.push({
+        op: 'add',
+        path: `${basePath}/hooks/afterUpdating`,
+        value: [ACMCrdHook, ACMImportHook],
+      });
+    } else {
+      if (!currentHooks.afterUpdating?.some(isAcmCrdHook)) {
+        patches.push({
```

Comment:
I think the flow here is wrong, because I had an initial Git config, and then I checked the checkbox.
It attempted to add two separate patches to `/spec/template/spec/hooks/afterUpdating`, rather than setting it to an array which includes both hooks.

It results in an error 400 because of it.

Reply:
with the `/-` at the end, it should work. But you're right that I could optimize this and do it in one patch for some cases.

Reply:
fixed